lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -141,10 +141,32 @@ def add_payment(post, payment, options)
       end
 
       def add_stored_credential(post, options)
-        return unless stored_credential = options[:stored_credential]
+        add_network_reference_id(post, options)
+        add_initiation_type(post, options)
+      end
+
+      def add_network_reference_id(post, options)
+        return unless options[:stored_credential] || options[:network_reference_id]
+
+        if options[:network_reference_id]
+          network_reference_id = options[:network_reference_id]
+        else
+          network_reference_id = options[:stored_credential][:network_transaction_id]
+        end
+
```

Comment:
Two things here: 
1. I would create this NTID override as options[:network_transaction_id] to keep it consistent across other gateways that have this override (even though Rapyd calls it something else)
2. I would use the ternary operator here. Something like 
`network_reference_id = options[:network_transaction_id] ? options[:network_transaction_id] : options[:stored_credential][:network_transaction_id]`.   That might be a little cleaner. You could also use this in the add_initiation_type method below.

Reply:
Thanks for the suggestions, it totally looks cleaner this way!

Reply:
I was running into rubocop errors when writing it `network_reference_id = options[:network_transaction_id] ? options[:network_transaction_id] : options[:stored_credential][:network_transaction_id]` as it's stating it was an unneeded conditional. The new way I wrote is the same as it'll prioritize the GSF if it's present and then default to the stored_credential hash. Also updated the tests and variable names to reflect network_transaction_id like you suggested and also added another assertion to the remote test when sending both stored credentials and options, thanks a bunch!